tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateStockFeatureContext.php

```diff
@@ -0,0 +1,325 @@
+        }
+
+        if (isset($data['out_of_stock_type'])) {
+            $command->setOutOfStockType($this->convertOutOfStockToInt($data['out_of_stock_type']));
+            unset($data['out_of_stock_type']);
+        }
+
+        if (isset($data['quantity'])) {
+            $command->setQuantity((int) $data['quantity']);
+            unset($data['quantity']);
+        }
+
+        if (isset($data['add_movement'])) {
+            $command->setAddMovement(PrimitiveUtils::castStringBooleanIntoBoolean($data['add_movement']));
+            unset($data['add_movement']);
+        }
+
+        if (isset($data['minimal_quantity'])) {
+            $command->setMinimalQuantity((int) $data['minimal_quantity']);
+            unset($data['minimal_quantity']);
+        }
+
+        if (isset($data['location'])) {
+            $command->setLocation($data['location']);
+            unset($data['location']);
+        }
+
+        if (isset($data['low_stock_threshold'])) {
+            $command->setLowStockThreshold((int) $data['low_stock_threshold']);
+            unset($data['low_stock_threshold']);
+        }
+
+        if (isset($data['low_stock_alert'])) {
+            $command->setLowStockAlert(PrimitiveUtils::castStringBooleanIntoBoolean($data['low_stock_alert']));
+            unset($data['low_stock_alert']);
+        }
+
+        if (isset($data['available_now_labels'])) {
+            $command->setLocalizedAvailableNowLabels($this->parseLocalizedArray($data['available_now_labels']));
+            unset($data['available_now_labels']);
+        }
+
+        if (isset($data['available_later_labels'])) {
+            $command->setLocalizedAvailableLaterLabels($this->parseLocalizedArray($data['available_later_labels']));
+            unset($data['available_later_labels']);
+        }
+
+        if (isset($data['available_date'])) {
+            $command->setAvailableDate(new DateTime($data['available_date']));
+            unset($data['available_date']);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param string $outOfStock
+     *
+     * @return int
+     */
+    private function convertOutOfStockToInt(string $outOfStock): int
+    {
+        $intValues = [
+            'default' => OutOfStockType::OUT_OF_STOCK_DEFAULT,
+            'available' => OutOfStockType::OUT_OF_STOCK_AVAILABLE,
+            'not_available' => OutOfStockType::OUT_OF_STOCK_NOT_AVAILABLE,
+            'invalid' => 42,
+        ];
+
+        return $intValues[$outOfStock];
+    }
+
+    /**
+     * @param string $outOfStock
+     *
+     * @return int
+     */
+    private function convertPackStockTypeToInt(string $outOfStock): int
+    {
+        $intValues = [
+            'default' => PackStockType::STOCK_TYPE_DEFAULT,
+            'products_only' => PackStockType::STOCK_TYPE_PRODUCTS_ONLY,
+            'pack_only' => PackStockType::STOCK_TYPE_PACK_ONLY,
+            'both' => PackStockType::STOCK_TYPE_BOTH,
+            'invalid' => 42,
```

Comment:
little comment explaining why its 42 and not 420 :smoking: :trollface: ?
I imagine smth like:
"This random number is hardcoded intentionally to reflect invalid stock type'

Reply:
Alright :trollface:
I see you are getting your revenge here 😛